### PR TITLE
Truncate feature

### PR DESCRIPTION
**Pull request by mikelococo**
See #60

### CODE REVIEW

File: layouts/_default/list.html [hunk 1]
```diff
@@ -31,7 +31,11 @@
 
 {{ $paginator := .Paginate (where .Data.Pages "Type" "post") }}
 {{ range $paginator.Pages }}
+{{ if (ne ($.Param "truncate") false) }}
```
Comment (mikelococo) [inline layouts/_default/list.html:34]: So I think I changed this such that it's settable either in config.tom or in post front-matter. After reading the "Param method" section of https://gohugo.io/templates/variables/, it doesn't seem like there should be anything to go wrong. I'm unable to verify the front-matter override in my manual tests, though.

* Setting truncate via the [params] section of config.toml works as expected (truncate if unset, truncate if true, full-content if false).
* Setting truncate via post front-matter seems to have no effect no matter what I do.

This is my first foray in to Go, Go templating, and Hugo theme development, and I'm at a bit of a loss how to debug this seemingly internal Hugo behavior. I'm posting the potentially broken PR in the hope that many eyeballs make bugs shallow.

Comment (alanorth) [reply layouts/_default/list.html:34]: You're right. This doesn't work when set in frontmatter—something about Page params not being available in the `range` context, or us not accessing them correctly. I don't have time to look at this more right now, but also, on second thought, I think it's better to just keep this simple and keep it as one site-wide setting. Sorry for making you do the extra work, but maybe we should just merge #59 instead.

**Comment by alanorth** [comment]
Ok, what I've done is just cherry-pick your first commit which checks the site-wide truncation setting in https://github.com/alanorth/hugo-theme-bootstrap4-blog/commit/2e5eef29a67c6d2b0ab4f11aac30a0b3645d4a40. So this is resolved and it's very simple, which is what I've always been trying to achieve in this theme. Thanks! 🎉

File: layouts/_default/list.html (hunk 1 above)
Comment (mikelococo) [reply layouts/_default/list.html:34]: I don't mind doing experiments in the hope of getting something better, even if it doesn't work out. I learned some good stuff as well. Thanks for the merge, and for your work maintaining this theme.

**Pull request closed without merging**